packages/workspace/src/workspace/nacl_files/nacl_files_source.ts

```
@@ -479,7 +481,7 @@ const buildNaclFilesSource = (
     }
     const naclFile = await getNaclFile(filename)
     if (naclFile === undefined) return undefined
-    return (await parseNaclFiles([naclFile], cache, functions))[0]
+    return (await parseNaclFiles([naclFile], (await getState()).parsedNaclFiles, functions))[0]
```

Not you (actually my fault :) but if we only get here when there's no state and thus no cache, you can use `parseNaclFile` and `toParsedNaclFile` here instead of `parseNaclFiles` with the array trick and the extra cache calls

---

I still prefer to use the regular method - since if in the future we will make changes for the way we use the cache in correlation with this we won't miss this spot... WdYT?

---

Looking at this again, how is it even working?

`await getState()` whilst `state === undefined` (and we only get here in this case) should throw an Error